internal/keys/datastore.go

```diff
@@ -0,0 +1,15 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package keys
```

Comment:
todo: Please use the singular for package name. `key`

Reply:
why?  Go has lots of packages with plural names (e.g. `strings`) and there is nothing in https://go.dev/blog/package-names against it.  `keys` contains many different key-types, and according to the norms I have experienced, the name should be plural.

Reply:
The packages in Go that use plural names are usually plural because the singular form is a reserved word or keyword (strings, types, errors, bytes).

It is generally preferred in the Go community to use singular names for packages. We've discussed this plenty of times before.

Reply:
Discussed over discord, and noted here for future reference:

When attempting to rename to `key`, I found this caused a large amount (+100) of variable shadowing (most places using this package want variables with the name `key`).  Even if consensus is reached RE avoiding plural package names, we probably still want this to be called `keys` anyway (quite possibly the same reason why many std lib packages have plural names).